gtsfm/frontend/detector_descriptor/superpoint.py

```diff
@@ -0,0 +1,78 @@
+"""Superpoint detector+descriptor implementation.
+
+The network was proposed in 'SuperPoint: Self-Supervised Interest Point Detection and Description' and is implemented
+by wrapping over the authors' implementation.
+
+References:
+- https://arxiv.org/abs/1712.07629
+- https://github.com/magicleap/SuperGluePretrainedNetwork
+
+Authors: Ayush Baid
+"""
+from pathlib import Path
+from typing import Tuple, Union
+
+import numpy as np
+import torch
+
+import gtsfm.utils.images as image_utils
+from gtsfm.common.image import Image
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.detector_descriptor.detector_descriptor_base import DetectorDescriptorBase
+from thirdparty.SuperGluePretrainedNetwork.models.superpoint import SuperPoint
+
+ROOT_PATH = Path(__file__).resolve().parent.parent.parent.parent
+MODEL_WEIGHTS_PATH = (
+    ROOT_PATH / "thirdparty" / "SuperGluePretrainedNetwork" / "models" / "weights" / "superpoint_v1.pth"
+)
```

Comment:
should this not be an option in the yaml file?

Reply:
So we expect the weights to be in a specific place, with our `download_model_weights.sh` script. So I think it's cleaner to put them in an expected place and use that location here.

It also allows us to use the authors' original SuperPoint API, without changing it at all